### PR TITLE
Document new strict-casts mode

### PR DESCRIPTION
**Pull request by srawlins**

### CODE REVIEW

File: resources/type-system/strict-casts.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  language:
+    strict-casts: true
+```
+
+## Motivation
+
+Static analysis can provide many benefits which require meaningful static
+types. For example, when a value of one type is _not assignable_ to a variable
+of another type, the error is reported by static analysis. However, when a
+value has a static type of `dynamic`, no error is reported; such a value is
```
Comment (munificent) [inline resources/type-system/strict-casts.md:28]: Grammar nit: I'd just use "." instead of ";". I've never found the semicolon to improve readability.

**Review by munificent — APPROVED** [review]
Drive-by review. This looks great!

File: resources/type-system/strict-casts.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+void main() {
+  var decoded = jsonDecode(/* some JSON */);
+  if (decoded['foo']) {
+    print('Got foo.');
+  }
+}
+```
+
+The expression used in an if-statement's condition must have a static type of
+`bool`. Without the Strict casts mode, an expression with a static type of
```
Comment (leafpetersen) [inline resources/type-system/strict-casts.md:43]: Strict -> strict

File: resources/type-system/strict-casts.md (hunk 1 above)
Comment (srawlins) [reply resources/type-system/strict-casts.md:28]: I love semicolons, and very probably overuse them :). Done.

File: resources/type-system/strict-casts.md (hunk 2 above)
Comment (srawlins) [reply resources/type-system/strict-casts.md:43]: done

**Pull request merged**